pkg/controller/postgressnapshot/postgressnapshot_controller.go

```diff
@@ -237,3 +245,52 @@ func (r *ReconcilePostgresSnapshot) createSnapshot(ctx context.Context, rdsSvc r
 	r.logger.Info(msg)
 	return croType.PhaseInProgress, croType.StatusMessage(msg), nil
 }
+
+func buildPostgresSnapshotInfoMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := buildPostgresSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+	if len(string(cr.Status.Phase)) != 0 {
+		labels["statusPhase"] = string(cr.Status.Phase)
+		return labels
+	}
+	return labels
+}
+
+func buildPostgresSnapshotGenericMetricLabels(cr *integreatlyv1alpha1.PostgresSnapshot, clusterID, snapshotName string) map[string]string {
+	labels := map[string]string{}
+	labels["clusterID"] = clusterID
+	labels["resourceID"] = cr.Name
+	labels["namespace"] = cr.Namespace
+	labels["instanceID"] = snapshotName
+	labels["productName"] = cr.Labels["productName"]
+	labels["strategy"] = postgresProviderName
+	labels["statusPhase"] = ""
+	return labels
+}
+
+func (r *ReconcilePostgresSnapshot) exposePostgresSnapshotMetrics(ctx context.Context, cr *integreatlyv1alpha1.PostgresSnapshot) {
+	// build instance name
+	snapshotName := cr.Status.SnapshotID
+
+	// get Cluster Id
+	logrus.Info("setting postgres snapshot information metric")
+	clusterID, err := resources.GetClusterID(ctx, r.client)
+	if err != nil {
+		logrus.Errorf("failed to get cluster id while exposing information metric for %v", snapshotName)
+		return
+	}
+
+	// build metric labels
+	infoLabels := buildPostgresSnapshotInfoMetricLabels(cr, clusterID, snapshotName)
+	// build generic labels
+	genericLabels := buildPostgresSnapshotGenericMetricLabels(cr, clusterID, snapshotName)
+
+	// set status gauge
+	resources.SetMetricCurrentTime(resources.DefaultPostgresSnapshotMetricName, infoLabels)
+
+	// set available metric
+	if len(string(cr.Status.Phase)) == 0 || cr.Status.Phase != croType.PhaseComplete {
+		resources.SetMetric(resources.DefaultPostgresSnapshotMetricName, genericLabels, 0)
+		return
+	}
+	resources.SetMetric(resources.DefaultPostgresSnapshotMetricName, infoLabels, 1)
```

Comment:
Because this metric is already being set on line `288` with a timestamp, I don't think we need this block here to set the values again to `0` or `1`

Reply:
We do need to handle the status if it hasn't been reconciled and status doesn't exist. maybe should be passing genericLabels on the last call to SetMetrics.

Reply:
shouldn't be setting this metric with timestamp, should only be 1 or 0 . will remove that line.